<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>MA42002 - 5&nbsp; Aggregation via chemotaxis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./DiffusionDrivenInstability.html" rel="next">
<link href="./LotkaVolteraPDE.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./LotkaVolteraPDE.html">Multi species</a></li><li class="breadcrumb-item"><a href="./BacterialChemotaxis.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Aggregation via chemotaxis</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">MA42002</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/dundeemath/MA42002/" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./MA42002.pdf" rel="" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./conservationequations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Conservation equations</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Single species</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./linearreactiondiffusion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Linear reaction diffusion equations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nonlinearreactiondiffusion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Non linear reaction diffusion equations</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Multi species</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./LotkaVolteraPDE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Lotka Voltera model</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./BacterialChemotaxis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Aggregation via chemotaxis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DiffusionDrivenInstability.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Diffusion driven instability</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SIR_PDE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Infectious disease</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NumericalMethods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Numerical methods in Python</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./linearstablityanalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Linear stability analysis of a system of nonlinear ODES</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#model-derivation" id="toc-model-derivation" class="nav-link active" data-scroll-target="#model-derivation"><span class="header-section-number">5.1</span> Model derivation</a></li>
  <li><a href="#numerical-solutions" id="toc-numerical-solutions" class="nav-link" data-scroll-target="#numerical-solutions"><span class="header-section-number">5.2</span> Numerical solutions</a></li>
  <li><a href="#spatially-homogeneous-steady-states" id="toc-spatially-homogeneous-steady-states" class="nav-link" data-scroll-target="#spatially-homogeneous-steady-states"><span class="header-section-number">5.3</span> Spatially Homogeneous Steady States</a></li>
  <li><a href="#stability-analysis" id="toc-stability-analysis" class="nav-link" data-scroll-target="#stability-analysis"><span class="header-section-number">5.4</span> Stability Analysis</a></li>
  <li><a href="#exercise" id="toc-exercise" class="nav-link" data-scroll-target="#exercise"><span class="header-section-number">5.5</span> Exercise</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Aggregation via chemotaxis</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Dictyostelium discoideum</strong> (Dicty) is a slime-mold that is widely studied experimentally as a model organism. The individual amoebae that constitute a slime-mold exhibit a range of phenomena also observed in mammalian cells e.g.&nbsp;differentiation, proliferation, migration.</p>
<p>Under nutirent starvation conditions, Dicty cells undergo complex collective behaviours. Individual amoebae secrete a diffusible chemical, cyclic AMP or cAMP. The amoebae respond chemotactically to cAMP and begin to migrate towards regions of high cAMP concentration via <em>chemotaxis</em>. As they migrate they generate a range of intricate patterns including <em>spiral waves</em> and <em>streaming</em> aggregation patterns (e.g. <a href="#fig-dicty_spiral">Figure&nbsp;<span>5.1</span></a>).</p>
<p>The movie below depicts slime mold aggregation. </p><div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/bkVhLJLG7ug" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div> Note the aggregation between 0:35 and 0.48.<p></p>
<p>In <a href="#fig-dicty_spiral">Figure&nbsp;<span>5.1</span></a> we can see stills-shot images of spiral patterns. This spirals are dynamic as can be seen in this movie. </p><div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/OX5Yiz38fgY" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div> See spiral pattern at around 0:36.<p></p>
<div id="fig-dicty_spiral" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="CG-14-355_F2.jpg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5.1: Spiral wave patterns underly Dictystelium aggregation. Image from <span class="citation" data-cites="durston2013dictyostelium">Durston (<a href="#ref-durston2013dictyostelium" role="doc-biblioref">2013</a>)</span>.</figcaption>
</figure>
</div>
<ul>
<li>can we simulate/model the mechanisms regulate that give rise to cellular aggregation?</li>
<li>how do features of patterns that form (e.g.&nbsp;pattern wavelength, speepd of aggregation) depend on individual cell properties?</li>
</ul>
<section id="model-derivation" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="model-derivation"><span class="header-section-number">5.1</span> Model derivation</h2>
<p>We consider a model for Dicty aggregation through the secretion of and chemotactic response to cAMP. We denote by <span class="math inline">\(n({\mathbf{x}}, t)\)</span> the density of amoebae and <span class="math inline">\(a({\mathbf{x}}, t)\)</span> the concentration of cAMP. The general conservation equation for the amoebae can be written:</p>
<p><span class="math display">\[
\frac{\partial n}{\partial  t} + \nabla \cdot {\mathbf{J}} = f(n,a),
\]</span> where <span class="math inline">\(f(n,a)\)</span> models any reaction terms for the amoebae e.g.&nbsp;proliferation, and the flux is given by <span class="math display">\[
{\mathbf{J}} = {\mathbf{J}}_{diffusion} + {\mathbf{J}}_{chemotaxis}.
\]</span></p>
<p>Assuming Fickian diffusion and the general chemotactic flux stated earlier (<a href="conservationequations.html#sec-conservation"><span>Section&nbsp;1.1</span></a>), the general <em>reaction-diffusion-chemotaxis</em> model for the amoebae responding to cAMP is given by:</p>
<p><span class="math display">\[
\begin{aligned}
\frac{\partial n}{\partial  t} &amp;=  \underbrace{D_n \nabla^2 n}_{diffusion} - \underbrace{\nabla \cdot \left( \chi(a) n \nabla a \right)}_{chemotaxis} + f(n,a),   \\
\frac{\partial a}{\partial  t} &amp; =   D_a \nabla^2 a + g(n,a),
\end{aligned}
\]</span></p>
<p>where we have assumed Fickian diffusion for the cAMP and <span class="math inline">\(g(a,n)\)</span> represents the kinetics i.e.&nbsp;source/sink terms, of cAMP.</p>
<p>One simple model has the following assumptiond:</p>
<p><span class="math display">\[f(n,a) = 0, \;\;\; g(n,a) = \mu n - \delta a, \;\;\; \chi (a) = \chi_0\]</span></p>
<p>i.e.&nbsp;</p>
<ul>
<li><p>there are no kinetics for the amoebae - they simply move randomly via diffusion and undergo chemotaxis in response to cAMP;</p></li>
<li><p>proliferation is neglected; - this is a reasonable assumption given the timescales involved, since they amoebae move on a faster timescale than they proliferate;</p></li>
<li><p>the amoebae are assumed to produce cAMP in proportion to their density, which means the more amoebae there are, the more cAMP (a reasonable first approximation);</p></li>
<li><p>the chemotactic function is taken to be a constant, again a reasonable first approximation;</p></li>
<li><p><span class="math inline">\(D_a &gt; D_n\)</span> since chemicals diffuse faster than cells move randomly.</p></li>
</ul>
<p>Under such assumptions we obtain the model equation</p>
<p><span class="math display">\[
\begin{aligned}
\frac{\partial n}{\partial  t} &amp; =  D_n \nabla^2 n - \chi_0 \nabla \cdot \left( n \nabla a \right), \\
\frac{\partial a}{\partial  t} &amp; =   D_a \nabla^2 a +  \mu n - \delta a,
\end{aligned}
\]</span></p>
<p>which becomes, upon considering a 1-dimensional domain <span class="math inline">\([0,L]\)</span>,</p>
<p><span id="eq-chemotaxis1d"><span class="math display">\[
\begin{aligned}
\frac{\partial n}{\partial  t} &amp;=  D_n \frac{\partial ^2 n}{\partial x^2} - \chi_0 \frac{\partial}{\partial x} \left( n \frac{\partial a}{\partial x} \right), \\
  &amp; &amp; \hspace{4.5cm} ( \dagger ) \\
\frac{\partial a}{\partial  t} &amp;=  D_a \frac{\partial ^2 a}{\partial x^2}  +  \mu n - \delta a,
\end{aligned}
\tag{5.1}\]</span></span></p>
<p>with zero flux boundary conditions:</p>
<p><span class="math display">\[
\begin{aligned}
D_a \frac{\partial a}{\partial  x} &amp; =  0, \;\;\; x = 0,L, \\
D_n \frac{\partial n}{\partial  x} - \chi_0 n \frac{\partial a}{\partial  x} &amp; =  0, \;\;\; x = 0,L.
\end{aligned}
\]</span></p>
<p>These reduce to:</p>
<p><span class="math display">\[
\frac{\partial a}{\partial  x} = \frac{\partial n}{\partial  x} = 0, \;\;\; x = 0,L.
\]</span></p>
</section>
<section id="numerical-solutions" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="numerical-solutions"><span class="header-section-number">5.2</span> Numerical solutions</h2>
<p>In <a href="#fig-bacterialchemotaxispde">Figure&nbsp;<span>5.2</span></a> we plot numerical solution of <a href="#eq-chemotaxis1d">Equation&nbsp;<span>5.1</span></a> together with no-flux boundary condition. The initial data are uniformly sampled. Note the emergence of periodic spatial structure in both variables. These correspond to peaks and troughs of cell density. The cells produce chemoattractant, <span class="math inline">\(a\)</span>, and the this induces a chemtactic flux up the gradient in <span class="math inline">\(a\)</span>. Hence more cells move towards regions where <span class="math inline">\(a\)</span> is high, more chemoattractant is produced in this region etc.</p>
<ul>
<li>What is the long-time behaviour of these solutions</li>
<li>For which parameters do we expect to see pattern formation?</li>
<li>How does spatial pattern depend on the initial data?</li>
</ul>
<div class="cell" data-execution_count="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.integrate <span class="im">import</span> odeint</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> random</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>T<span class="op">=</span><span class="dv">80</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>L<span class="op">=</span><span class="dv">150</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>mu<span class="op">=</span><span class="fl">1.2</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>delta<span class="op">=</span><span class="fl">0.1</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>D_n<span class="op">=</span><span class="fl">2.50</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>D_a<span class="op">=</span><span class="fl">2.5</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>chi_0<span class="op">=</span><span class="fl">1.4</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>N_x<span class="op">=</span><span class="dv">200</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>N_t<span class="op">=</span><span class="dv">100</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>t<span class="op">=</span>np.linspace(<span class="dv">1</span>,T,N_t)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>x<span class="op">=</span>np.linspace(<span class="dv">0</span>,L,N_x)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>u_0<span class="op">=</span>np.ones_like(x)<span class="op">+</span><span class="fl">0.01</span><span class="op">*</span>np.random.uniform(low<span class="op">=</span><span class="fl">0.0</span>, high<span class="op">=</span><span class="fl">0.1</span>, size<span class="op">=</span>(N_x,))</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>n_0<span class="op">=</span>np.ones_like(x)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>u_0<span class="op">=</span>np.concatenate((u_0,n_0))</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>dx<span class="op">=</span>L<span class="op">/</span>(N_x<span class="op">-</span><span class="dv">1</span>)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>dt<span class="op">=</span>T<span class="op">/</span>(N_t<span class="op">-</span><span class="dv">1</span>)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> LVPDErhs(sol,t):</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    N_x<span class="op">=</span><span class="bu">int</span>(np.ceil(<span class="bu">len</span>(sol)<span class="op">/</span><span class="dv">2</span>))</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    n<span class="op">=</span>sol[<span class="dv">0</span>:N_x]</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    a<span class="op">=</span>sol[N_x:<span class="dv">2</span><span class="op">*</span>N_x]</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>    f_n<span class="op">=</span>np.zeros_like(n)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>    f_a<span class="op">=</span>np.zeros_like(n)</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>,N_x<span class="op">-</span><span class="dv">2</span>):</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>      f_n[i]<span class="op">=</span>D_n<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(n[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span><span class="dv">2</span><span class="op">*</span>n[i]<span class="op">+</span>n[i<span class="op">+</span><span class="dv">1</span>]) <span class="op">-</span> chi_0<span class="op">*</span>n[i]<span class="op">*</span><span class="dv">1</span><span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span><span class="dv">2</span><span class="op">*</span>a[i]<span class="op">+</span>a[i<span class="op">+</span><span class="dv">1</span>])<span class="op">-</span>chi_0<span class="op">/</span>(<span class="dv">2</span><span class="op">*</span>dx)<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">+</span><span class="dv">1</span>]<span class="op">-</span>a[i<span class="op">-</span><span class="dv">1</span>])<span class="op">*</span>(n[i<span class="op">+</span><span class="dv">1</span>]<span class="op">-</span>n[i<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>    i<span class="op">=</span><span class="dv">0</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>    f_n[i]<span class="op">=</span>D_n<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(<span class="op">-</span>n[i]<span class="op">+</span>n[i<span class="op">+</span><span class="dv">1</span>]) <span class="op">-</span> chi_0<span class="op">*</span>n[i]<span class="op">*</span><span class="dv">1</span><span class="op">/</span>(<span class="dv">2</span><span class="op">*</span>dx)<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(<span class="op">-</span>a[i]<span class="op">+</span>a[i<span class="op">+</span><span class="dv">1</span>])<span class="op">-</span>chi_0<span class="op">/</span>(<span class="dv">2</span><span class="op">*</span>dx)<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">+</span><span class="dv">1</span>]<span class="op">-</span>a[i])<span class="op">*</span>(n[i<span class="op">+</span><span class="dv">1</span>]<span class="op">-</span>n[i])</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>    i<span class="op">=</span>N_x<span class="op">-</span><span class="dv">1</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>    f_n[i]<span class="op">=</span>D_n<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(n[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span>n[i])<span class="op">-</span> chi_0<span class="op">*</span>n[i]<span class="op">*</span><span class="dv">1</span><span class="op">/</span>(<span class="dv">2</span><span class="op">*</span>dx)<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span>a[i])<span class="op">-</span>chi_0<span class="op">/</span>(<span class="dv">2</span><span class="op">*</span>dx)<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i]<span class="op">-</span>a[i<span class="op">-</span><span class="dv">1</span>])<span class="op">*</span>(n[i]<span class="op">-</span>n[i<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>,N_x<span class="op">-</span><span class="dv">2</span>):</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>      f_a[i]<span class="op">=</span>D_a<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span><span class="dv">2</span><span class="op">*</span>a[i]<span class="op">+</span>a[i<span class="op">+</span><span class="dv">1</span>]) </span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>    i<span class="op">=</span><span class="dv">0</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>    f_a[i]<span class="op">=</span>D_a<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(<span class="op">-</span>a[i]<span class="op">+</span>a[i<span class="op">+</span><span class="dv">1</span>]) </span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>    i<span class="op">=</span>N_x<span class="op">-</span><span class="dv">1</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>    f_a[i]<span class="op">=</span>D_a<span class="op">/</span>dx<span class="op">**</span><span class="dv">2</span><span class="op">*</span>(a[i<span class="op">-</span><span class="dv">1</span>]<span class="op">-</span>a[i])</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>    reaction_n<span class="op">=</span><span class="dv">0</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>    reaction_a<span class="op">=</span>mu<span class="op">*</span>n<span class="op">-</span>delta<span class="op">*</span>a</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    f_n<span class="op">=</span>f_n<span class="op">+</span>reaction_n</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    f_a<span class="op">=</span>f_a<span class="op">+</span>reaction_a</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>    f<span class="op">=</span> np.concatenate((f_n, f_a)) </span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> f  </span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>sol<span class="op">=</span>odeint(LVPDErhs,u_0,t)</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>n<span class="op">=</span>sol[:,<span class="dv">0</span>:N_x]</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>a<span class="op">=</span>sol[:,<span class="dv">0</span>:N_x]</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>fig, ax <span class="op">=</span> plt.subplots(<span class="dv">2</span>,<span class="dv">1</span>)</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].plot(x,n[<span class="dv">0</span>,:],<span class="st">'r'</span>)</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].plot(x,n[<span class="dv">16</span>,:],<span class="st">'b'</span>)</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].plot(x,n[<span class="dv">32</span>,:],<span class="st">'m'</span>)</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].plot(x,n[<span class="dv">48</span>,:],<span class="st">'k'</span>)</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].set_xlabel(<span class="st">'$x$'</span>)</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">0</span>].set_ylabel(<span class="st">'$n$'</span>)</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].plot(x, a[<span class="dv">0</span>,:],<span class="st">'r--'</span>)</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].plot(x, a[<span class="dv">16</span>,:],<span class="st">'b--'</span>)</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].plot(x, a[<span class="dv">32</span>,:],<span class="st">'m--'</span>)</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].plot(x, a[<span class="dv">48</span>,:],<span class="st">'k--'</span>)</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].set_xlabel(<span class="st">'$x$'</span>)</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>ax[<span class="dv">1</span>].set_ylabel(<span class="st">'$a$'</span>)</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>plt.legend([<span class="st">'t'</span><span class="op">+</span> <span class="bu">str</span>(t[<span class="dv">0</span>]),<span class="st">'t='</span><span class="op">+</span> <span class="bu">str</span>(t[<span class="dv">4</span>]),<span class="st">'t='</span><span class="op">+</span> <span class="bu">str</span>(t[<span class="dv">8</span>]),<span class="st">'t='</span><span class="op">+</span> <span class="bu">str</span>(t[<span class="dv">12</span>])])</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">'$x$'</span>)</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>plt.grid()</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div id="fig-bacterialchemotaxispde" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="BacterialChemotaxis_files/figure-html/fig-bacterialchemotaxispde-output-1.png" width="614" height="429" class="figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5.2: Numerical solution of bacterial chemtaxis model</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="spatially-homogeneous-steady-states" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="spatially-homogeneous-steady-states"><span class="header-section-number">5.3</span> Spatially Homogeneous Steady States</h2>
<p>Define the total number of cells, <span class="math inline">\(N\)</span>, <span class="math display">\[
N= \int_0^L n(x,t)dx
\]</span> Upon differentiation with respect to time <span class="math display">\[
\begin{aligned}
\frac{dN}{dt} &amp;=\int_0^L \frac{\partial n(x,t)}{\partial t}dx
&amp;=\int_0^L  D_n \frac{\partial ^2 n}{\partial x^2} - \chi_0 \frac{\partial}{\partial x} \left( n \frac{\partial a}{\partial x} \right) dx.
\end{aligned}
\]</span> Upon integration of the right-hand side w.r.t. <span class="math inline">\(x\)</span>, subsequent application of the no-flux boundary condition implies <span class="math display">\[
\frac{dN}{dt}=0.
\]</span> Hence the total number of cells in the domain is fixed by the initial data. For a spatially homogeneoud solutions, th4 cells must be uniformly distributed in space, i.e. <span class="math display">\[
n^*=N/L
\]</span></p>
<p>Additionaly, the spatially homogeneous steady state solution <span class="math inline">\((n^* , a^* )\)</span> satisfies:</p>
<p><span class="math display">\[
a^*=\frac{\mu}{\delta} n^*.
\]</span></p>
<p>We now undertake a linear stability analysis to determine whether this is stable or unstable. If the above spatially homogeneous steady state is unstable, this will indicate that aggregation patterns may arise in the system.</p>
</section>
<section id="stability-analysis" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="stability-analysis"><span class="header-section-number">5.4</span> Stability Analysis</h2>
<p>In a similar manner to previous stability analyses, we consider small perturbations around the spatially homogeneous steady state <span class="math inline">\((n^* , a^* )\)</span>, i.e.</p>
<p><span class="math display">\[
n(x,t) = n^* + \tilde{n}(x,t), \;\;\; a(x,t) = a^* + \tilde{a}(x,t)
\]</span></p>
<p>where <span class="math inline">\(\tilde{n}(x,t)\)</span> and <span class="math inline">\(\tilde{a}(x,t)\)</span> are “small” so that higher order terms can be neglected.</p>
<p><strong>NOTE</strong> Unlike previous stability analysis, these perturbations are both <em>time</em> and <em>space</em> dependent.</p>
<p>Substituting the above perturbations into equations <span class="math inline">\(( \dagger )\)</span>, we neglect higher order terms and retain only linear terms. This is largely straightforward, but we provide some detail for the linearization of the chemotactic term i.e.&nbsp;</p>
<p><span class="math display">\[
\frac{\partial}{\partial x} \left[ ( n^* + \tilde{n}) \frac{\partial}{\partial x} \left( a^* + \tilde{a} \right) \right] = \frac{\partial}{\partial x} \left[ ( n^* + \tilde{n}) \frac{\partial \tilde{a}}{\partial x}  \right] \approx n^* \frac{\partial ^2 \tilde{a}}{\partial x^2}.
\]</span></p>
<p>The fully linearized system is then given by: <span id="eq-chemotax_linear"><span class="math display">\[
\begin{aligned}
\frac{\partial \tilde{n}}{\partial  t} &amp; =  D_n \frac{\partial ^2 \tilde{n}}{\partial x^2} - \chi_0 n^* \frac{\partial ^2 \tilde{a}}{\partial x^2} \\
\frac{\partial \tilde{a}}{\partial  t} &amp; =   D_a \frac{\partial ^2 \tilde{a}}{\partial x^2}  +  \mu \tilde{n} - \delta \tilde{a}.
\end{aligned}
\tag{5.2}\]</span></span></p>
<p>Although the above equations are linear, an explicit solution is non-trivial and we are required to make a further “separation of variables” <em>ansatz</em>. We seek solutions of the form <span class="math display">\[
\tilde{n}(t,x) = u(t) \phi_1(x)
\]</span> and <span class="math display">\[
\tilde a (t,x) = v(t) \phi_2(x)
\]</span></p>
<p>Upon substitution <span class="math display">\[
\begin{aligned}
&amp;\frac{\partial u}{\partial  t}  \phi_1  =   D_n  u\frac{\partial ^2 \phi_1}{\partial x^2} - \chi_0 n^*  v\frac{\partial ^2 \phi_1}{\partial x^2} \\
&amp; \frac{\partial v}{\partial  t}  \phi_2=     D_a v \frac{\partial ^2  \phi_2}{\partial x^2}  +  \mu u \,  \phi_1 - \delta v\,  \phi_2, \\
\end{aligned}
\]</span> with boundary conditions <span class="math display">\[
\begin{aligned}
&amp; u \frac{\partial \phi_1}{\partial x} = 0, \quad   v \frac{\partial \phi_2}{\partial x} = 0 \quad \text{ for } \; x = 0, \; x=L.
\end{aligned}
\]</span></p>
<p>We assume that<br>
<span class="math display">\[
\phi_1 = \phi_2 = \phi,
\]</span> where <span class="math inline">\(\phi\)</span> is the solution of the elliptic problem <span class="math display">\[
\begin{aligned}
\frac{d^2 \phi}{dx^2} &amp;= - k^2 \phi &amp;&amp; \text{ in } \; (0,L), \\
\frac{d \phi}{dx} &amp;= 0  &amp;&amp; \text{ for } \; x=0, \; x=L.
\end{aligned}
\]</span> We can compute that solution of the equation for <span class="math inline">\(\phi\)</span> are of the form <span class="math display">\[
\phi(x) = A \cos(kx) + B\sin(kx).
\]</span> Since <span class="math inline">\(\phi\)</span> satisfied zero Neumann boundary conditions we have that <span class="math display">\[
\phi(x) = A \cos(kx),
\]</span> where <span class="math inline">\(A\)</span> is an arbitrary constanr, and <span class="math display">\[
k = \dfrac {m \pi} L, \quad m \in \mathbb N.
\]</span></p>
<p>Then we have <span class="math display">\[
\begin{aligned}
&amp;\frac{\partial u}{\partial  t}  \phi  =   - k^2 D_n  u \phi +  \chi_0 n^* k^2\,  v \, \phi,  \\
&amp; \frac{\partial v}{\partial  t} \phi =   - k^2 D_a  \, v \phi  +  \mu u \,  \phi - \delta v\,  \phi,
\end{aligned}
\]</span> and since <span class="math inline">\(\phi\)</span> is not identically zero on <span class="math inline">\((0,L)\)</span> we obtain a system of linear ODEs for <span class="math inline">\((u(t),v(t))\)</span> <span class="math display">\[
\begin{aligned}
&amp;\frac{\partial u}{\partial  t}    =   - k^2 D_n  u  +  \chi_0 n^* k^2\,  v,  \\
&amp; \frac{\partial v}{\partial  t} =   - k^2 D_a  \, v  +  \mu u  - \delta v.
\end{aligned}
\]</span></p>
<p>We know that solutions of linear ODEs have the form <span class="math display">\[
u(t) = C_1 e^{\lambda t} \quad \textrm{and} \quad v(t) = C_2 e^{\lambda t}
\]</span> for some constant <span class="math inline">\(C_1\)</span>, <span class="math inline">\(C_2\)</span> and <span class="math inline">\(\lambda\)</span> are eigenvalues of the corresponding matrix.</p>
<p>Thus we obtain <span class="math display">\[
\begin{aligned}
\lambda C_1 &amp; =  - D_n k^2 C_1  + \chi_0 n^* k^2 C_2,  \\
\lambda C_2 &amp; = - D_a k^2 C_2 +  \mu C_1 - \delta C_2,
\end{aligned}
\]</span> which can be written</p>
<p><span class="math display">\[
\left(
\begin{array}{cc}
- D_n k^2 - \lambda &amp;  \chi_0 n^* k^2 \\
\mu &amp; - D_a k^2 -\delta -\lambda
\end{array}
\right)
\left(
\begin{array}{c}
C_1 \\
C_2
\end{array}
\right) = \mathbf{0} .
\]</span></p>
<p><strong>Remark</strong> Notice that we obtained that <span class="math inline">\(\tilde n\)</span> and <span class="math inline">\(\tilde a\)</span> are of the form <span class="math display">\[\tilde{n}(x,t) = C_1 e^{\lambda t} e^{ikx}, \;\;\; \tilde{a}(x,t) = C_2 e^{\lambda t} e^{ikx}.\]</span></p>
<p>For a non-trivial solution (for non-trivial perturbations <span class="math inline">\(\tilde n\)</span>, <span class="math inline">\(\tilde a\)</span>), i.e.&nbsp;<span class="math inline">\(C_1 \neq 0\)</span> and <span class="math inline">\(C_2 \neq 0\)</span>, the determinant of the above matrix must be zero, and this leads to the following quadratic equation to be solved for <span class="math inline">\(\lambda\)</span>:</p>
<p><span class="math display">\[
\lambda^2 + \left( D_n k^2 + D_a k^2 + \delta \right) \lambda + D_n k^2 \left( D_a k^2 + \delta \right) - \mu \chi_0 n^* k^2 = 0.
\]</span></p>
<p>This is of the form <span class="math display">\[
\lambda^2 + \alpha \lambda + \beta = 0,
\]</span></p>
<p>and so has roots: <span class="math display">\[
\lambda = \frac{-\alpha \pm \sqrt{\alpha^2 - 4 \beta}}{2}.\]</span></p>
<p><strong>NOTE</strong> This has two <em>real</em> roots, since <span class="math display">\[
\alpha^2 - 4 \beta &gt; 0
\]</span> (see Exercise/Tutorial).</p>
<p>For stability, we require both roots to be negative. Since both roots are real, this leads to:</p>
<p><span class="math display">\[
\lambda &lt; 0  \Leftrightarrow \alpha &gt; 0 \;\;\; \text{and} \;\;\; \beta &gt;0.
\]</span></p>
<p>Now <span class="math display">\[\alpha = D_n k^2 + D_a k^2 + \delta &gt; 0,
\]</span> and so for stability, we require <span class="math inline">\(\beta &gt; 0\)</span> i.e.&nbsp;</p>
<p><span class="math display">\[
\begin{aligned}
&amp; &amp; D_n k^2 \left( D_a k^2 + \delta \right) - \mu \chi_0 n^* k^2 &gt; 0 \\
&amp; &amp; \Rightarrow \mu \chi_0 n^* &lt;  D_n  \left( D_a k^2 + \delta \right)
\end{aligned}
\]</span> Hence, we will have instability when this condition is not satisfied i.e.&nbsp; <span class="math display">\[
\mu \chi_0 n^* &gt;  D_n  \left( D_a k^2 + \delta \right).
\]</span></p>
<p>The precise value of <span class="math inline">\(k^2\)</span> can be determined from the zero-flux boundary conditions i.e. <span class="math display">\[
k  =  \frac{m \pi}{L}, \;\;\; m = 1,2, \dots
\]</span> if we look for non-constant <span class="math inline">\(\phi\)</span>.</p>
<p>Hence, we will have instability whenever <span class="math display">\[
\mu \chi_0 n^* &gt;  D_n  \left( D_a \frac{m^2 \pi^2}{L^2} + \delta \right), \;\;\; m = 1,2, \dots
\]</span></p>
<p>It can be shown (see Exercise/Tutorial), that <span class="math inline">\(\lambda (k^2)\)</span> ( or <span class="math inline">\(\lambda (m^2)\)</span>) is monotonic decreasing and hence the fastest growing mode is <span class="math inline">\(m=1\)</span> i.e.&nbsp;we have an instability as long as <span class="math display">\[
\mu \chi_0 n^* &gt;  D_n  \left( \frac{D_a \pi^2}{L^2} + \delta \right).
\]</span></p>
<p>In general, from the above inequality, we can say that there is a likelihood of instability (amoebae aggregation) if:</p>
<ul>
<li><span class="math inline">\(D_a\)</span>, <span class="math inline">\(D_n\)</span> and <span class="math inline">\(\delta\)</span> are all ``small’’</li>
<li>L is ``large’’</li>
<li><span class="math inline">\(\chi_0 , \mu , n^*\)</span> are ``large’’</li>
</ul>
<p>Considering all other parameters to be fixed, in theory the above result states that it is possible to find a large enough value for the chemotactic coefficient <span class="math inline">\(\chi_0\)</span> to satisfy the instability condition i.e.&nbsp;chemotaxis induces instability and leads to aggregation of the amoebae.</p>
</section>
<section id="exercise" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="exercise"><span class="header-section-number">5.5</span> Exercise</h2>
<p>From the results we have obtained we deduce that:</p>
<ul>
<li>chemotaxis has a destabilizing effect</li>
<li>diffusion has a stabilizing effect on spatially homogeneous solutions</li>
</ul>
<p>If this is true then one might expect the numerical results presented in <a href="#fig-bacterialchemotaxispde">Figure&nbsp;<span>5.2</span></a> to have spatially homogeneous solutions if the diffusion coefficient is made sufficiently large. * Can you test this by running the code for larger values of the parameter <span class="math inline">\(D\)</span>? * Alternatively, what happens if you make the chemotactic coefficient <span class="math inline">\(\chi_0\)</span> smaller? * what kind of aggregation patterns do you see if the system is solved in two spatial dimensions?</p>
<p>However, there is one type of system where diffusion also has a destabilizing effect…</p>


<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-durston2013dictyostelium" class="csl-entry" role="listitem">
Durston, AJ. 2013. <span>“Dictyostelium: The Mathematician’s Organism.”</span> <em>Current Genomics</em> 14 (6): 355–60.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./LotkaVolteraPDE.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Lotka Voltera model</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./DiffusionDrivenInstability.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Diffusion driven instability</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>